From 59b46dd4b4522d331c4758a0a5a467c5cd0de89c Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@mailbox.org>
Date: Mon, 23 Feb 2026 15:06:46 +0100
Subject: [PATCH] Revert "enhance: make openssl dependency shared (#47664)"

This reverts commit 0dbc6dccc655e7fd31372fe7acae0d97e073ef47.

The build process currently builds various libraries typically provided
by the system package manager, some of them as shared objects and others
as static archives. Since the build process also sets a LD_LIBRARY_PATH
that includes the freshly built artifacts, ld will start mixing shared
objects provided by the system and by Milvus--also for tools of the
build process itself. This can result in incompatible library versions
at build time, e.g. cmake from the system using a new libcurl from the
system in combination with an old libssl from the build:

    cmake: /home/build/.conan/data/openssl/3.1.2/_/_/package/815539f297fca2b9743bcf1cc21c4d0ffb36be68/lib/libssl.so.3: version `OPENSSL_3.2.0' not found (required by /lib/libcurl.so.4)

Temporarily fix this by building OpenSSL as a static library again, as
OpenSSL is the one dependency most likely to already be provided by the
system and having version conflicts.

Signed-off-by: Markus Boehme <markus.boehme@mailbox.org>
---
 internal/core/conanfile.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/internal/core/conanfile.py b/internal/core/conanfile.py
index 02684f6c5d..2b194c910f 100644
--- a/internal/core/conanfile.py
+++ b/internal/core/conanfile.py
@@ -49,7 +49,6 @@ class MilvusConan(ConanFile):
 
     generators = ("cmake", "cmake_find_package")
     default_options = {
-        "openssl:shared": True,
         "libevent:shared": True,
         "double-conversion:shared": True,
         "folly:shared": True,
-- 
2.52.0

